gatsby-plugin-material-ui/src/gatsby-node.js

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+
+// Write out the gatsby-mui-config module to .cache.
+// Happens before webpack starts compiling page.
+// .cache is available when gatsby-browser.js and gatsby-ssr.js runs.
+
+// eslint-disable-next-line import/prefer-default-export
+export const onPreBootstrap = ({ store }, pluginOptions) => {
+  const { program } = store.getState();
+
+  const fixPath = module => {
+    let newModule = module;
+    if (os.platform() === `win32`) {
+      newModule = module.split(`\\`).join(`\\\\`);
+    }
+    return newModule;
+  };
+
+  let module;
+
+  // Not allowed to specify both pathToStylesProvider and StylesProvider props
+  // in gatsby-config.js
+  try {
+    if (pluginOptions.pathToStylesProvider && pluginOptions.stylesProvider) {
+      throw new Error(
+        `You specified both pathToStylesProvider and stylesProvider in gatsby-config.js. Only one should be specified...`,
+      );
+    }
+
+    if (pluginOptions.stylesProvider) {
+      console.log(
+        `pluginOptions.stylesProvider:`,
+        pluginOptions.stylesProvider,
+      );
+      module = `const stylesProviderProps = ${JSON.stringify(
+        pluginOptions.stylesProvider,
+      )};
+
+export default stylesProviderProps;
+`;
+      console.log(`module:`, module);
+    } else if (pluginOptions.pathToStylesProvider) {
+      module = `import stylesProviderProps from "${
+        path.isAbsolute(pluginOptions.pathToStylesProvider)
+          ? pluginOptions.pathToStylesProvider
+          : path.join(program.directory, pluginOptions.pathToStylesProvider)
+      }";
+export default stylesProviderProps;
+`;
+
+      module = fixPath(module);
+    } else {
+      module = `const stylesProviderProps = {
```

Comment:
Many use the plugin without options. Is there a reason why you changed the default behavior?

Reply:
No reason that I remember right now. injectFirst by default?

Reply:
There is no reason for a breaking change here. Previously, the plugin simply returned the element if no options were passed
```javascript
if (pluginOptions.stylesProvider) {
    return (
      <StylesProvider {...pluginOptions.stylesProvider}>
        {element}
      </StylesProvider>
    );
  }

  return element;
```

Reply:
@hupe1980 I modified the ssr as well, to include the conditional (so it corresponds to browser). I will change it back, if that was wrong.

Reply:
You need to do something else with server-side rendering. In your implementation, styles are no longer collected when no options are specified.

Reply:
@hupe1980 I now removed the conditionals I previously introduced in the SSR. I am not sure why the conditionals should be in `gatsby-browser.js`, but not in `gatsby-browser.js`. However, if it works now I will leave it as it is.

Reply:
On the server side, the styles must always be collected and later integrated into the page as a style tag:
```js
setHeadComponents([
      <style
        id="jss-server-side"
        key="jss-server-side"
        dangerouslySetInnerHTML={{ __html: css }}
      />,
    ]);
```
On the browser side, this task does not exist.

Without conditionals, the server side is configured differently than the browser page. 

What about this?

```js
//gatsby-node.js

else {
      module = `const stylesProviderProps = {}; // empty! 
export default stylesProviderProps;
`;
    }
```